### ISSUE TITLE
Performance drop when schema has "uniqueItems" set to true

### ISSUE THREAD

**Issue by NitinRamesh25** [santhosh-tekuri/jsonschema#121]
Hi,

I noticed a major performance drop when the schema has `uniqueItems` set as true for an array and the json content being a 4mb file has a massive array.

The benchmark test took only about 0.07260 ns/op when set to false and a whopping 107824033000 ns/op (~107 sec) when set to true.

Another library from [xeipuuv/gojsonschema](https://github.com/xeipuuv/gojsonschema) seems to not have any major time difference when uniqueItems is set to true. So perhaps there is room for improvement here.

The schema used is as follows:
```json
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "type": "array",
  "items": {
    "type": "object",
    "required": ["name", "price", "description"],
    "properties": {
      "name": {
        "type": "string"
      },
      "price": {
        "type": "number"
      },
      "description": {
        "type": "string"
      }
    }
  },
  "uniqueItems": true
}
```

**Comment by santhosh-tekuri** [comment]
fixed with caf0c40

@NitinRamesh25 can you verify with this fix

**Comment by NitinRamesh25** [comment]
@santhosh-tekuri I have verified with the new change and it works fast when duplicate items are close by in the array.
In case where the first and last element of the large array are duplicates, then it still takes a lot of time.
I would prefer if it runs fast even when no duplicate items are found.

Wouldn't [this](https://github.com/santhosh-tekuri/jsonschema/blob/master/schema.go#L425) be faster if we marshal the element to []byte, convert it to a string and store it as the key of a map ?
Since the array elements would be keys of the map, lookups for duplicate data would be faster.

**Comment by NitinRamesh25** [comment]
Much thanks. It's faster now.

**Comment by santhosh-tekuri** [comment]
@NitinRamesh25  just published v5.3.1 with this fix
